File: algorithms/comparison/bubblesort.py
def bubble_sort(arr: list[int]) -> list[int]:
    """Bubble sort: a naive sorting algorithm that compares and swaps two adjacent values

    Args:
        arr (list): a non-sorted list of integers

    Returns:
        list: a sorted list of integers in increasing order
    """
    n = len(arr)
    for i in range(n):
        for j in range(i, n):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]
    return arr

File: algorithms/comparison/cpp_version/bubble_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

std::vector<int> bubble_sort(std::vector<int> arr)
{
    int n = arr.size();

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                std::swap(arr[j], arr[j + 1]);
            }
        }
    }
    return arr;
}

std::vector<int> main()
{
    std::vector<int> arr;
    bubble_sort(arr);
}